packages/fuels-test-helpers/src/fuel_bin_service.rs

```
@@ -0,0 +1,222 @@
+impl FuelService {
+    pub async fn new_node(config: Config) -> FuelResult<Self> {
+        let requested_port = config.addr.port();
+
+        let bound_address = match requested_port {
+            0 => get_socket_address(),
+            _ if is_free(requested_port) => config.addr,
+            _ => return Err(Error::IOError(std::io::ErrorKind::AddrInUse.into())),
+        };
+
+        let config = Config {
+            addr: bound_address,
+            ..config
+        };
+
+        let params = ServerParams {
+            extended_config: ExtendedConfig {
+                config,
+                config_file: NamedTempFile::new()?,
+            },
+        };
+
+        let runner = run_node(params).await?;
+
+        Ok(FuelService {
+            bound_address,
+            runner: Box::pin(runner),
+        })
+    }
+
+    pub async fn stop(&self) -> FuelResult<State> {
+        self.runner.abort();
+        Ok(State::Stopped)
+    }
+}
+
+async fn server_health_check(client: &FuelClient) -> FuelResult<()> {
+    let mut attempts = 5;
+    let mut healthy = client.health().await.unwrap_or(false);
+    let between_attempts = Duration::from_millis(300);
+
+    while attempts > 0 && !healthy {
+        healthy = client.health().await.unwrap_or(false);
+        sleep(between_attempts).await;
+        attempts -= 1;
+    }
+
+    if !healthy {
+        panic!("error: Could not connect to fuel core server.")
+    }
+
+    Ok(())
+}
+
+fn get_socket_address() -> SocketAddr {
+    let free_port = pick_unused_port().expect("No ports free");
+    SocketAddr::new("127.0.0.1".parse().unwrap(), free_port)
+}
+
+async fn run_node(params: ServerParams) -> FuelResult<JoinHandle<()>> {
+    let ServerParams {
+        mut extended_config,
+    } = params;
+    let args = extended_config.config_to_args_vec()?;
+
+    let binary_name = "fuel-core";
+    let paths = which::which_all(binary_name)
+        .unwrap_or_else(|_| panic!("failed to list '{binary_name}' binaries"))
```

IMO we should return an error. `or_else` could be used here with our error